server/util/scratchspace/scratchspace.go

```diff
@@ -63,5 +63,5 @@ func tempDir() string {
 //	    os.Remove(f.Name())
 //	}()
 func CreateTemp(pattern string) (*os.File, error) {
-	return os.CreateTemp(tempDir(), pattern)
+	return os.CreateTemp(TempDir(), pattern)
```

Comment:
imo it would be better to use `scratchspace.CreateTemp()` as the top level dir, then create whatever dirs you want under that. That way we don't ever have to worry about naming collisions and we don't have to expose a new public API.

If you wanted a fixed directory name so that you could inspect it locally, you could still inspect it using something like

`cd /tmp/buildbuddy/template-repo-*/ && git status (etc...)`

Reply:
I tried this out, but `scratchspace.CreateTemp()` creates a new temporary file rather than creating a new temporary directory.

Reply:
We can add `scratchspace.MkdirTemp()` func (as the analogue of `os.MkdirTemp`)

Reply:
Done